packages/assets-controllers/src/RatesController/RatesController.ts

```diff
@@ -0,0 +1,186 @@
+import { BaseController } from '@metamask/base-controller';
+import { Mutex } from 'async-mutex';
+
+import { fetchMultiExchangeRate as defaultFetchExchangeRate } from '../crypto-compare-service';
+import type {
+  RatesMessenger,
+  ConversionRates,
+  RatesControllerState,
+  RatesControllerOptions,
+} from './types';
+
+export const name = 'RatesController';
+
+const DEFAULT_CURRENCIES = {
+  btc: 'btc',
+};
+
+const DEFAULT_INTERVAL = 180000;
+
+const metadata = {
+  currency: { persist: true, anonymous: true },
+  rates: { persist: true, anonymous: true },
+  fromCurrencies: { persist: true, anonymous: true },
+};
+
+const defaultState = {
+  currency: 'usd',
+  rates: {
+    btc: {
+      conversionDate: 0,
+      conversionRate: 0,
+      usdConversionRate: null,
+    },
+  },
+  fromCurrencies: [DEFAULT_CURRENCIES.btc],
+};
+
+export class RatesController extends BaseController<
+  typeof name,
+  RatesControllerState,
+  RatesMessenger
+> {
+  readonly #mutex = new Mutex();
+
+  readonly #fetchMultiExchangeRate;
+
+  readonly #includeUsdRate;
+
+  #intervalId: NodeJS.Timeout | undefined;
+
+  #intervalLength: number | undefined;
```

Comment:
Yes, because these variables don't have a set value when the class is being instantiated.

Reply:
Yes for `#intervalId` but we do define a value for `#internvalLength` in the constructor, hence this should be a `number` since it will always be "defined".

I've check the compiler behavior using this: https://www.typescriptlang.org/play/?#code/MYGwhgzhAEBiD28BGYBO0DeAoa0DEYAcgK4C2SApqgFzQB2ZlqO0A9K9AJIAu0AFpGjd40StAAGAH2jE6AEwoAzAJZ0Kc8fyoUANNAirgFaAHdjaY3Xi9IBgOZ1Vd6GDoBPaADcwIYhRbs0KpCfMbA8HQQ3KjEwMKoAIQBHIQA8gAqAKK0nIrQbvDE0KgUpPCextyhEtKyCipqGtAADmjcelVh8KTNyiBUpn0g0OE94KosBACyYG6UAKrySqrqJORUtAzr6LVLDepYh7jhkdGx8QAUAJSYLLhVyhAAdARrTNAAvNAADADcLABfI7FChgOTXW64XAlbjEVB0aAAbQez1ejCoHT4jxeYBmcwoi3qKzkbyoAF1-rggQCgA